Framework/Source/Graphics/Scene/SceneRenderer.cpp

```diff
@@ -278,37 +287,24 @@ namespace Falcor
             {
                 draw(currentData, pMesh, activeInstances);
             }
+
+            // Restore the program state
+            if (pMesh->hasBones())
+            {
+                pProgram->removeDefine("_VERTEX_BLENDING");
+            }
         }
     }
 
     void SceneRenderer::renderModelInstance(CurrentWorkingData& currentData, const Scene::ModelInstance* pModelInstance)
     {
-        const Model* pModel = pModelInstance->getObject().get();
+        mpLastMaterial = nullptr;
 
-        if (setPerModelData(currentData))
+        // Loop over the meshes
+        for (uint32_t meshID = 0; meshID < pModelInstance->getObject()->getMeshCount(); meshID++)
         {
-            Program* pProgram = currentData.pState->getProgram().get();
```

Comment:
Ha? What happened to all that code?

Reply:
`setPerModelData` was  moved to `renderScene` where the model loop is. `_VERTEX_BLENDING` define was moved to happen per mesh in `renderMeshInstances` because all meshes are not necessarily influenced by bone(s). Everything else is indentation changes.